packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNode.ts

```diff
@@ -193,24 +194,117 @@ export class SummarizerNode implements IRootSummarizerNode {
+		);
+		assert(this.wipReferenceSequenceNumber !== undefined, "Not tracking a summary");
+
+		// If the parent node skipped recursion, it did not call summarize on this node. So, summarize was not missed
+		// but was intentionally not called.
+		// Otherwise, summarize should have been called on this node and wipLocalPaths must be set.
+		if (parentSkipRecursion || this.wipLocalPaths !== undefined) {
+			return false;
+		}
+
+		/**
+		 * The absence of wip local path indicates that summarize was not called for this node. Return failure.
+		 * This can happen if:
+		 * 1. A child node was created after summarize was already called on the parent. For example, a data store
+		 * is realized (loaded) after summarize was called on it creating summarizer nodes for its DDSes. In this case,
+		 * parentSkipRecursion will be true and the if block above would handle it.
+		 * 2. A new node was created but summarize was never called on it. This can mean that the summary that is
+		 * generated may not have the data from this node. We should not continue, log and throw an error. This
+		 * will help us identify these cases and take appropriate action.
+		 *
+		 * This happens due to scenarios such as data store created during summarize. Such errors should go away when
+		 * summarize is attempted again.
+		 */
+		return true;
+	}
+
+	/**
+	 * Called after summary has been uploaded to the server. Add the work-in-progress state to the pending summary
+	 * queue. We track this until we get an ack from the server for this summary.
+	 * @param proposalHandle - The handle of the summary that was uploaded to the server.
 	 */
-	public completeSummary(proposalHandle: string) {
-		this.completeSummaryCore(proposalHandle, undefined, false);
+	public completeSummary(proposalHandle: string, validate: boolean) {
+		this.completeSummaryCore(
+			proposalHandle,
+			undefined /* parentPath */,
+			false /* parentSkipRecursion */,
+			validate,
+		);
 	}
 
 	/**
-	 * Recursive implementation for completeSummary, with additional internal-only parameters
+	 * Recursive implementation for completeSummary, with additional internal-only parameters.
+	 * @param proposalHandle - The handle of the summary that was uploaded to the server.
+	 * @param parentPath - The path of the parent node which is used to build the path of this node.
+	 * @param parentSkipRecursion - true if the parent of this node skipped recursing the child nodes when summarizing.
+	 * In that case, the children will not have work-in-progress state.
+	 * @param validate - true to validate that the in-progress summary is correct for all nodes.
 	 */
 	protected completeSummaryCore(
 		proposalHandle: string,
 		parentPath: EscapedPath | undefined,
 		parentSkipRecursion: boolean,
+		validate: boolean,
 	) {
-		assert(
-			this.wipSummaryLogger !== undefined,
-			0x1a3 /* "wipSummaryLogger should have been set in startSummary or ctor" */,
-		);
+		if (validate && this.wasSummarizeMissed(parentSkipRecursion)) {
```

Comment:
@agarwal-navin - This validation used to be skipped if `parentSkipRecursion` and `this._latestSummary === undefined`. Due to early return.  Is it ok to change it?

Reply:
As you pointed out in chat - I'm wrong, `wasSummarizeMissed` returns false if `parentSkipRecursion`